add profile, map, login, registration components

nikita-ivanov-93 opened the pull request:
Добавил компоненты приложения: карта, профиль, форма логина, форма регистрации, Навигация при помощи стейта.
Добавил header с кнопками навигации.
Настроил commitizen.

nikita-ivanov-93 committed:
feat: add profile, map, login, registration components

Добавил компоненты приложения, навигация при помощи стейта.

isaevfeed commented on src/Login.jsx:
```diff
@@ -0,0 +1,47 @@
+        </div>
+      </form>
+      <section>
+        {Pages[this.state.currentPage]}
```
Очень странно, что эта логика у тебя находится в компоненте login, который должен отвечать только за страницу логина. Мне кажется это вместе со стейтом и navigatTo нужно вынести в App

nikita-ivanov-93 replied:
Убрал лишнюю логику из компонентов login и registration, оставил толькоо в Аpp

isaevfeed commented on src/Registration.jsx:
```diff
@@ -0,0 +1,46 @@
+import Login from './Login';
+import {Map} from './Map';
+
+const Pages = {
```
Сейчас из компонентов Login, Register и др это лучше вынести в App. Но на будущее, если у тебя какая-то константа или переменная общего типа используется более 1-го раза, то ее нужно выносить в отдельный переиспользуемый файл и импортировать

nikita-ivanov-93 replied:
got it

nikita-ivanov-93 committed:
perf: same operations were deleted

Удалил одинаковые операции из формы логина и регистрации, оставил только в app.js

nikita-ivanov-93 committed:
fix: correct import of  registration and login components

Исправил импорт компонентов логина и регистрации, так как теперь это компоненты - функции
